src/mongocxx/options/aggregate.cpp

```diff
@@ -33,6 +36,8 @@ aggregate& aggregate::batch_size(std::int32_t batch_size) {
 }
 
 aggregate& aggregate::max_time(std::chrono::milliseconds max_time) {
+    if (!is_int32_duration<std::chrono::milliseconds>(max_time))
```

Comment:
You actually shouldn't need to write the template type here. It should be able to be inferred from the argument.